test/controllers/AccountControllerSpec.scala

```diff
@@ -0,0 +1,125 @@
+import org.joda.time.DateTime
+import org.mockito.BDDMockito.given
+import org.mockito.Matchers._
+import org.scalatest.mock.MockitoSugar
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Helpers}
+import play.filters.csrf.CSRF.SignedTokenProvider
+import uk.gov.hmrc.crypto.Protected
+import uk.gov.hmrc.play.frontend.auth.AuthenticationProvider
+import uk.gov.hmrc.play.http.{HeaderCarrier, SessionKeys}
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+import scala.concurrent.Future
+
+class AccountControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplication {
+
+  Helpers.running(fakeApplication) {
+
+    trait Setup {
+      val underTest = new AccountController {
+        val authConnector = mock[AuthConnector]
+
+        def authProvider = mock[AuthenticationProvider]
+      }
+
+      implicit val encryptedStringFormats = JsonStringEncryption
+      implicit val decryptedStringFormats = JsonStringDecryption
+      implicit val format = Json.format[LoginDetails]
+
+      val csrfToken = "csrfToken" -> SignedTokenProvider.generateToken
+      val authToken = SessionKeys.authToken -> "some-bearer-token"
+      val userToken = GatekeeperSessionKeys.LoggedInUser -> "userName"
+
+      val aLoggedInRequest = FakeRequest().withSession(csrfToken, authToken, userToken)
+      val aLoggedOutRequest = FakeRequest().withSession(csrfToken)
+
+    }
+
+    "loginPage" should {
+      "be loaded with not authenticated user" in new Setup {
+        val result = await(underTest.loginPage()(aLoggedOutRequest))
+        status(result) shouldBe 200
+      }
+
+      "be skipped with an authenticated user" in new Setup {
+        val result = await(underTest.loginPage()(aLoggedInRequest))
+        status(result) shouldBe 303
+      }
+    }
+
+    "authenticateAction" should {
+      "go to dashboard and set cookie when user authenticated successfully" in new Setup {
+        val loginDetails = LoginDetails("userName", Protected("password"))
+        val aValidFormJson = Json.toJson(loginDetails)
+        val successfulAuthentication = SuccessfulAuthentication(BearerToken("bearer-token", DateTime.now().plusMinutes(10)), "userName", None)
+
+        given(underTest.authConnector.login(any[LoginDetails])(any[HeaderCarrier])).willReturn(Future.successful(successfulAuthentication))
+
+        val result = await(underTest.authenticateAction()(
+          aLoggedOutRequest.withJsonBody(aValidFormJson)))
+        status(result) shouldBe 303
+        result.header.headers should contain("Location" -> "/api-gatekeeper/dashboard")
+        result.header.headers.get("Set-Cookie") shouldBe defined
+
+        session(result).get(SessionKeys.authToken) shouldBe Some("bearer-token")
+
+        val cookie = result.header.headers.get("Set-Cookie").get
+        cookie should include("mdtp=")
+        cookie should include("authToken=bearer-token")
```

Comment:
From the test, it's not obvious that the authToken is *contained* in the mdtp cookie - it looks like it could be its own cookie.  

I wonder if there's some way to turn this Set-Cookie into a datastructure that makes it clearer that the mdtp cookie contains the authToken?